Source/WebCore/page/SecurityOrigin.cpp

```diff
@@ -553,8 +559,8 @@ Ref<SecurityOrigin> SecurityOrigin::create(const String& protocol, const String&
 {
     String decodedHost = PAL::decodeURLEscapeSequences(host);
     auto origin = create(URL { protocol + "://" + host + "/" });
-    if (port && !WTF::isDefaultPortForProtocol(*port, protocol))
-        origin->m_data.port = port;
+    if (port && !WTF::isDefaultPortForProtocol(*port, protocol) && std::holds_alternative<SecurityOriginData::Tuple>(origin->m_data.data))
```

Comment:
This new code looks a lot more complicated, for the same functionality. This is not a good compromise.
While using a variant is a good idea in theory, the execution here is not quite right in my opinion.

SecurityOrigin shouldn't need to know anything about the internals of SecurityOriginData, this is bad encapsulation. It used to be OK because SecurityOriginData was a simple struct. However, now that it uses a variant internally and has getters, I don't think it is still OK for SecurityOrigin to go poke inside SecurityOriginData's internals like this.